docs/DeveloperGuide.md

```diff
@@ -1,34 +1,514 @@
+* `RTemplate` - Resistor Circuit Template
+* `RcTemplate` - Resistor-Capacitor Template (extends `RTemplate`)
+* `LrTemplate` - Inductor-Resistor Template (extends `RTemplate`)
+
+![CircuitCommandClass](diagrams/CircuitCommandClassDiagram.png)
+
+<small><i>Figure 6</i></small>
+
+The diagram above demonstrates the relationship between the various `CircuitCommand` objects. The various commands to be parsed are as explained in this section. While the **User Guide** explains the commands used on the CLI, this section goes into detail the classes used to execute the commands. The command classes also make use of the classes in the [**`Model`**](#model-component) shown in the diagram below when executing the commands.
+
+![CircuitModelClass](diagrams/CircuitModelClassDiagram.png)
+
+<small><i>Figure 7</i></small>
+
+### `TemplateCircuitCommand`  <a name="temp-circ"></a>
+Given below is the sequence diagram for interactions within the `logic` and `template` components for the 
+`parse(template rl)` API call that implements the `template` command to create templates.
+
+![TemplateSequence](diagrams/TemplateSequenceDiagram.png)
+
+<small><i>Figure 8</i></small>
+
+### `SetCircuitCommand` <a name="set-circ"></a>
+
+Given below is the Sequence Diagram for interactions within the `Logic` component for the `parse("set r 500")` API call which implements the `set` command to set values of components.
+
+![SetSequence](diagrams/SetSequenceDiagram.png)
+
+<small><i>Figure 9</i></small>  
+
+### `AddCircuitCommand` <a name="add-circ"></a>
+
+Given below is the detailed Sequence diagram for interactions within the `logic`, `template` and `component` components 
+for the `parse("add parallel c 20")` API call that implements the `add` command to add components.
+
+![AddSequence](diagrams/AddSequenceDiagram.png)
+
+<small><i>Figure 10</i></small>  
+
+### `CalculateCircuitCommand` <a name="calc-circ"></a>
+The calculate command can be split into two distinct sequence diagrams. Given below is the sequence diagram for the
+calculate command that does not show the access of the `component` component and shows the interactions for the
+`parse("calc power")` API call that implements this version of the `calculate` command to calculate power.  
+
+![CalcPowerSequence](diagrams/CalcPowerSequenceDiagram.png)
+
+<small><i>Figure 11</i></small>  
+
+The second sequence diagram given below shows the detailed interaction that acesses the `component` class through the
+`parse("calc reff")` API call that implements this version of the `calculate` command to calculate effective resistance.
+
+![CalcReffSequence](diagrams/CalcReffSequenceDiagram.png)
+
+<small><i>Figure 12</i></small>  
+
+--------------------------------------------------------------------------------------------------------------------
+
+## Implementation of Boolean Commands (Praveen) <a name="bool-comd"></a>
+
+![BooleanCommandClass](diagrams/BooleanCommandClassDiagram.png)
+
+<small><i>Figure 13</i></small>
```

Comment:
Hi, I think having a top down hierarchy would improve readability.

Reply:
Yep, in AY2021S1-CS2113T-W13-3#121!